pwiz_tools/Skyline/Model/Proteome/ProteinAssociation.cs

```diff
@@ -805,24 +807,61 @@ public SrmDocument CreateDocTree(SrmDocument current, IProgressMonitor monitor)
             var proteinAssociationsList = ParsimoniousProteins.OrderBy(kvp => kvp.Key.RecordIndex).ToList();
 
             var newPeptideGroups = new List<PeptideGroupDocNode>(); // all groups that will be added in the new document
-            //var assignedPeptides = proteinAssocationsList.SelectMany(kvp => kvp.Value).ToHashSet();
+            var appendPeptideLists = new List<PeptideGroupDocNode>();
+
+            // Move unmapped peptides from FastaSequence node to "Unmapped Peptides" list
+            var unmappedPeptideNodes = _peptideToPath.SelectMany(kvp => kvp.Value).Where(p => !p.IsDecoy).ToHashSet();
+            unmappedPeptideNodes.ExceptWith(ParsimoniousProteins.Values.SelectMany(pag => pag.Peptides));
 
             // Modifies and adds old groups that still contain unmatched peptides to newPeptideGroups
             foreach (var nodePepGroup in current.MoleculeGroups)
             {
-                // Get non-peptide children
-                var nonPeptideNodes = nodePepGroup.Children.Where(node => (node as PeptideDocNode)?.Peptide.Target.IsProteomic == false).ToList();
+                var peptideDocNodes = nodePepGroup.Children.Where(node => node is PeptideDocNode).Cast<PeptideDocNode>().ToList();
 
-                // Ignore old groups with no non-peptide children
-                if (nonPeptideNodes.Count == 0)
+                // drop all empty nodes
+                if (peptideDocNodes.Count == 0)
+                    continue;

```

Comment:
I think this should be:
`appendPeptideLists.Add(nodePepGroup);`
Any peptide list which was empty before they did "Associate Proteins" should be ignored and remain in the document.